amps/ags/rm-automation/rm-automation-community-rest-api/src/test/java/org/alfresco/rest/rm/community/hold/AddToHoldsV1Tests.java

```diff
@@ -0,0 +1,135 @@
+import static org.alfresco.utility.report.log.Step.STEP;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.alfresco.dataprep.CMISUtil;
+import org.alfresco.dataprep.ContentActions;
+import org.alfresco.rest.model.RestNodeAssociationModelCollection;
+import org.alfresco.rest.model.RestNodeModel;
+import org.alfresco.rest.rm.community.base.BaseRMRestTest;
+import org.alfresco.rest.rm.community.model.hold.Hold;
+import org.alfresco.rest.rm.community.model.hold.HoldChild;
+import org.alfresco.rest.rm.community.model.user.UserRoles;
+import org.alfresco.rest.rm.community.requests.gscore.api.FilePlanAPI;
+import org.alfresco.rest.v0.service.RoleService;
+import org.alfresco.utility.constants.UserRole;
+import org.alfresco.utility.model.FileModel;
+import org.alfresco.utility.model.SiteModel;
+import org.alfresco.utility.model.UserModel;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * V1 API tests for adding content/record folder/records to holds
+ *
+ * @author Damian Ujma
+ */
+public class AddToHoldsV1Tests extends BaseRMRestTest
+{
+    private static final String HOLD = "HOLD" + generateTestPrefix(AddToHoldsV1Tests.class);
+    private String holdNodeRef;
+    private FileModel documentHeld;
+    private final List<UserModel> users = new ArrayList<>();
+
+    @Autowired
+    private RoleService roleService;
+    @Autowired
+    private ContentActions contentActions;
+
+    @BeforeClass(alwaysRun = true)
+    public void preconditionForAddContentToHold()
+    {
+        STEP("Create a hold.");
+        Hold hold = createHold(FILE_PLAN_ALIAS, Hold.builder().name(HOLD).description("Description").reason("No reason").build(), getAdminUser());
+        holdNodeRef = hold.getId();
+        STEP("Create test files.");
+        SiteModel testSite = dataSite.usingAdmin().createPublicRandomSite();
+        documentHeld = dataContent.usingAdmin().usingSite(testSite)
+            .createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+        dataContent.usingAdmin().usingSite(testSite)
+            .createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+        dataContent.usingAdmin().usingSite(testSite)
+            .createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+
+        STEP("Add the content to the hold.");
+        getRestAPIFactory()
+            .getHoldsAPI(getAdminUser())
+            .addChildToHold(HoldChild.builder().id(documentHeld.getNodeRefWithoutVersion()).build(), hold.getId());
+
+        STEP("Create users");
+        UserModel userAddHoldPermission = roleService.createUserWithSiteRoleRMRoleAndPermission(testSite,
+            UserRole.SiteCollaborator, holdNodeRef, UserRoles.ROLE_RM_MANAGER, PERMISSION_FILING);
+        users.add(userAddHoldPermission);
```

Comment:
I don't understand what this list is for.

Reply:
It will be required for more integration tests. In this PR I added only basic IT tests. For now it's not needed so I removed it.